### PR TITLE
Persist internal debug logging setting

### PR DESCRIPTION
**Pull request by nshuba**
Task/Issue URL: https://app.asana.com/0/0/1205391076687195/f

### Description
* Persist debug logging setting for internal users
* Disable NetGuard logs for internal users, unless they have debug logging set

### Steps to test this PR
- [x] Install an internal _release_ version from this branch
- [x] Turn on AppTP and check that no VPN logs appear (e.g. `tag:NetGuard.JNI` and `message:"VPN log"`)
- [x] Go to AppTP Dev Settings and enable VPN Debug Logging
- [x] Check that NetGuard logs now appear
- [x] Disable and re-enable AppTP - make sure logs still appear
- [x] Force stop the app
- [x] Open and re-enable AppTP - make sure logs still appear and VPN Debug Logging is still set to ON
- [x] Disable VPN Debug Logging
- [x] Check that no VPN logs appear (e.g. `tag:NetGuard.JNI` and `message:"VPN log"`)

### CODE REVIEW

File: app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSetting.kt [hunk 1]
```diff
@@ -22,6 +22,7 @@ enum class AppTpSetting(override val value: String, override val defaultValue: B
     ProtectGames("protectGames"),
     ExceptionLists("exceptionLists", defaultValue = true),
     RestartOnConnectivityLoss("restartOnConnectivityLoss", defaultValue = true),
+    VerboseLogging("verboseLogging"),
```
Comment (aitorvs) [inline app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSetting.kt:25]: We're trying to deprecate these `AppTpSetting`s (deprecated way to do feature flagging).
 Can we use a feature flag created with `@ContributesRemoteFeature` instead?
You could add it in `:vpn-impl` under `com.duckduckgo.mobile.android.vpn.debug` package. No need to expose it in an `:api` module as `:vpn-internal` has visibility to all `:vpn-impl`

```kotlin
// This feature will never be in remote config
@ContributesRemoteFeature(
    scope = AppScope::class,
    featureName = "appTpLocalFeature",
)
interface AppTpLocalFeature {
    @Toggle.DefaultValue(false)
    fun self(): Toggle

    @Toggle.DefaultValue(false)
    fun verboseLogging(): Toggle
}
```

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt [hunk 2]
```diff
@@ -57,6 +58,7 @@ class NgVpnNetworkStack @Inject constructor(
     private val runtime: Runtime,
     private val appTrackerDetector: AppTrackerDetector,
     private val trackingProtectionAppsRepository: TrackingProtectionAppsRepository,
+    private val appTpFeatureConfig: AppTpFeatureConfig,
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt:61]: This would then become 
```suggestion
    private val appTpLocalFeature: AppTpLocalFeature
```

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt [hunk 3]
```diff
@@ -180,7 +182,7 @@ class NgVpnNetworkStack @Inject constructor(
         val vpnNetwork = vpnNetwork.safeGet().getOrElse { return Result.failure(it) }
         if (tunnelThread == null) {
             logcat { "Start native runtime" }
-            val level = if (appBuildConfig.isDebug || appBuildConfig.isInternalBuild()) VpnNetworkLog.DEBUG else VpnNetworkLog.ASSERT
+            val level = if (appBuildConfig.isDebug || appTpFeatureConfig.isEnabled(AppTpSetting.VerboseLogging)) VpnNetworkLog.DEBUG else VpnNetworkLog.ASSERT
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt:185]: This would then become
```suggestion
            val level = if (appBuildConfig.isDebug || appTpLocalFeature.verboseLogging().isEnabled()) VpnNetworkLog.DEBUG else VpnNetworkLog.ASSERT
```

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/logging/AndroidLogcatLoggerRegistrar.kt [hunk 4]
```diff
@@ -38,14 +40,15 @@ import logcat.logcat
...
+    private val appTpFeatureConfig: AppTpFeatureConfig,
 ) : VpnProcessLifecycleObserver, MainProcessLifecycleObserver {
 
     override fun onCreate(owner: LifecycleOwner) {
         onVpnProcessCreated()
     }
 
     override fun onVpnProcessCreated() {
-        if (appBuildConfig.isDebug) {
+        if (appBuildConfig.isDebug || appTpFeatureConfig.isEnabled(AppTpSetting.VerboseLogging)) {
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/logging/AndroidLogcatLoggerRegistrar.kt:51]: Ditto
```suggestion
        if (appBuildConfig.isDebug || appTpLocalFeature.verboseLogging().isEnabled()) {
```

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/logging/AndroidLogcatLoggerRegistrar.kt [hunk 5]
```diff
@@ -38,14 +40,15 @@ import logcat.logcat
 )
 class AndroidLogcatLoggerRegistrar @Inject constructor(
     private val appBuildConfig: AppBuildConfig,
+    private val appTpFeatureConfig: AppTpFeatureConfig,
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/logging/AndroidLogcatLoggerRegistrar.kt:43]: Ditto
```suggestion
    private val appTpLocalFeature: AppTpLocalFeature,
```

File: app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/AppTpFeatureConfigImplTest.kt [hunk 6]
```diff
@@ -63,6 +63,7 @@ class AppTpFeatureConfigImplTest {
                 AppTpSetting.ProtectGames -> assertFalse(config.isEnabled(setting))
                 AppTpSetting.ExceptionLists -> assertTrue(config.isEnabled(setting))
                 AppTpSetting.RestartOnConnectivityLoss -> assertTrue(config.isEnabled(setting))
+                AppTpSetting.VerboseLogging -> assertFalse(config.isEnabled(setting))
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/AppTpFeatureConfigImplTest.kt:66]: Ditto. We could remove this

File: app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSettingTest.kt [hunk 7]
```diff
@@ -31,6 +31,7 @@ class AppTpSettingTest {
                 AppTpSetting.ProtectGames -> assertFalse(setting.defaultValue)
                 AppTpSetting.ExceptionLists -> assertTrue(setting.defaultValue)
                 AppTpSetting.RestartOnConnectivityLoss -> assertTrue(setting.defaultValue)
+                AppTpSetting.VerboseLogging -> assertFalse(setting.defaultValue)
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSettingTest.kt:34]: Ditto. We could remove this

File: app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt [hunk 8]
```diff
@@ -51,6 +52,7 @@ class NgVpnNetworkStackTest {
     private val runtime: Runtime = mock()
     private val appTrackerDetector: AppTrackerDetector = mock()
     private val trackingProtectionAppsRepository: TrackingProtectionAppsRepository = mock()
+    private val appTpFeatureConfig: AppTpFeatureConfig = mock()
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt:55]: Ditto.
You could mock it but also create a fake, see `FakeNetPRemoteFeatureFactory` for an example

File: app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSetting.kt (hunk 1 above)
Comment (nshuba) [reply app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/feature/AppTpSetting.kt]: Cool, that's much cleaner, thanks!

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt [hunk 9]
```diff
@@ -64,7 +64,7 @@ class NetpRealAppTrackerDetector constructor(
         val packageId = appNameResolver.getPackageIdForUid(uid) ?: return null
 
         if (VpnExclusionList.isDdgApp(packageId) || packageId.isInExclusionList()) {
-            logcat { "shouldAllowDomain: $packageId is excluded, allowing $domain" }
+            logcat { "shouldAllowDomain: $packageId is excluded, allowing packet" }
```
Comment (aitorvs) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt:67]: Why the change?

**Review by aitorvs — COMMENTED** [review]
left one comment. Otherwise lgtm

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt (hunk 9 above)
Comment (nshuba) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt:67]: When I was writing up privacy triage I realized that if our internal users exclude an app, they may not want the app's domains to show up in the logs. Especially since we sometimes ask our internal users to send us logs for extra debugging.

Comment (aitorvs) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt:67]: Not sure I agree with that. Those logs are primarily for us dev in debug mode.
If internal users don't want to send the logs they just don't. Also, these logs are not enabled in prod or internal builds. 

Not sure what privacy triage has to do here tbf.

Comment (nshuba) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/app/tracking/NetpRealAppTrackerDetector.kt:67]: We had an offline conversation about privacy triage being needed or not. Regardless, the triage got approved and I would prefer to keep this change if you don't mind.

**Review by aitorvs — APPROVED** [review]
